skyportal/handlers/api/comment.py

```diff
@@ -48,9 +57,29 @@ def get(self, comment_id):
               application/json:
                 schema: Error
         """
-        comment = Comment.get_if_accessible_by(
-            comment_id, self.current_user, raise_if_none=True
-        )
+
+        if comment_on_what is None:
+            comment_on_what = 'object'
+
+        if type(comment_on_what) != str:
+            return self.error(
+                f'Wrong input type {type(comment_on_what)} given as "comment_on_what" argument'
+            )
+
+        if comment_on_what.lower() == "object":  # comment on object (default)
+            comment = Comment.get_if_accessible_by(
+                comment_id, self.current_user, raise_if_none=True
+            )
+        elif comment_on_what.lower() == "spectrum":
+            comment = CommentOnSpectrum.get_if_accessible_by(
+                comment_id, self.current_user, raise_if_none=True
+            )
+        # add more options using elif
+        else:
+            return self.error(
+                f'Wrong input "{comment_on_what}" given as "comment_on_what" argument.'
```

Comment:
```suggestion
                f'Unsupported input "{comment_on_what}" given as "comment_on_what" argument.'
```
I think this would just be a bit clearer